reports/sb108/sb108.qmd

```diff
@@ -343,94 +355,116 @@ Of the **`r nrow(parties) |> format(big.mark = ",")` total rows / charges** in t
 
 parties_sum |>
   filter(n >= 2) |>
-  ggplot(aes(x = n)) +
-    # scale_x_discrete() +
-    stat_count() +
-    geom_text(aes(label = after_stat(count)), stat = "count", vjust = -0.2)
+  ggplot(aes(
+    x = n_capped
+  )) +
+  scale_y_continuous(labels = scales::comma) +
+  stat_count() +
+  geom_text(aes(label = format(after_stat(count), big.mark = ",")), stat = "count", vjust = -0.2) +
+  labs(title = "Simple possession charges per defendant",
+       subtitle = "All CF / CM simple possession charges\nfiled 2001-2022",
+       x = "N simple possession charges in data",
+       y = "Total defendants",
+       caption = "This data is based on the total number\nof charges in our dataset, not cases")
 
 ```
 
-Next, we'll look at how many of the charges only involved marijuana, as they would be excluded from SB108:
+### How many **non-marijuana** simple possession charges per defendant?
+
+Next, we'll repeat this same analysis with marijuana-specific charges removed, as they would be excluded from SB108:
 
 ```{r}
-parties |>
+
+parties_sum_no_marijuana <- parties_clean |>
+  filter(!marijuana) |>
+  group_by(oscn_id) |>
+  summarize(
+    n = n(),
+    n_capped = if_else(n >= 4, "4+", as.character(n)),
+    list_charges = paste0(count_as_disposed, collapse = "; ")
+  ) |>
+  ungroup()
+
+parties_sum_no_marijuana |>
+  filter(n >= 2) |>
+  ggplot(aes(
+    x = n_capped, 
+  )) +
+  scale_y_continuous(labels = scales::comma) +
+  stat_count() +
+  geom_text(aes(label = format(after_stat(count), big.mark = ",")), stat = "count", vjust = -0.2) +
+  labs(title = "Non-marijuana simple possession charges\nper defendant",
+       subtitle = "Non-marijuana CF / CM simple possession charges\nfiled 2001-2022",
+       x = "N non-marijuana possession charges in data",
+       y = "Total defendants",
+       caption = "This data is based on the total number\nof charges in our dataset, not cases")
+
+```
+
+### How many defendants have 4+ **non-marijuana** simple possession charges in a ten-year timespan?
+
+Finally, SB108 stipulates that the felony charge only comes into effect after 3+ simple possession **convictions** in a **ten year timespan**. Therefore, we'll filter out charges that ended in a dismissal (leaving only convictions and deferred / suspended dispositions) and repeat the analysis with an added 10-year rolling timeframe.
+
+```{r}
+poi <- parties |>
+  arrange(oscn_id, disposition_date) |>
+  group_by(oscn_id) |>
+  nest() |>
   mutate(
-    marijuana = str_detect(
-      count_as_disposed,
-      regex("\\bMA.*NA\\b|\\bMJ\\b|\\bMARI\\b", ignore_case = T)
-    ),
-    other = str_detect(
-      count_as_disposed,
-      "(?i)OXY|HUFF|AMPHET|ZOLOL|ZOLAM|HYDROC|CODEIN|PRECURS|XANAX|MORPH|METERDI|ZEPAM|LORAZ|VALIU|EPHED|COCA|PSEUDO|\\bPRESCRIP|\\bNARC|\\bMETH|HEROIN|ANHYD|AMMONIA|OPIUM|LORTAB|\\bPARAPH\\b|\\bOTHER\\b"
+    rolling_10_year_n = map_int(
+      data, # Shouldn't this be `d` so that we're only checking the ones we care about?
```

Comment:
This is where I think the issue is. From what I can tell, it's checking all of each person's charges (because `data` hasn't been filtered or anything, it's just the raw data pull), not only the ones we're interested in. That ends up giving us a number that's too high because it includes stuff that has nothing to do with simple possession. However, I didn't have enough time to actually figure out how it should look.

Reply:
@brancengregory